x/ccv/provider/types/msg.go

```diff
@@ -2,18 +2,27 @@ package types
 
 import (
 	"encoding/json"
+	"fmt"
 	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
 )
 
 // provider message types
 const (
 	TypeMsgAssignConsumerKey           = "assign_consumer_key"
 	TypeMsgRegisterConsumerRewardDenom = "register_consumer_reward_denom"
+	TypeMsgSubmitConsumerMisbehaviour  = "submit_consumer_misbehaviour"
+	TypeMsgSubmitConsumerDoubleVoting  = "submit_consumer_double_vote"
 )
 
-var _ sdk.Msg = &MsgAssignConsumerKey{}
+var (
+	_ sdk.Msg = &MsgAssignConsumerKey{}
+	_ sdk.Msg = &MsgSubmitConsumerMisbehaviour{}
```

Comment:
similar to Philip's [comment above](https://github.com/cosmos/interchain-security/pull/1287/files#r1324318529) ... do we need `MsgSubmitConsumerDoubleVoting` here?

Reply:
Good catch